egograph/pipelines/infrastructure/notification/service.py

```
@@ -0,0 +1,94 @@
+    GenericAdapter,
+    WebhookAdapter,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _custom_message_for(exc: Exception | None) -> str | None:
+    """例外型からユーザー/エージェント向けメッセージを組み立てる。
+
+    新たな例外型に対応するときはここに ``isinstance`` 分岐を追加するだけで
+    済む（Open/Closed）。
```

Ruff (0.15.15)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)